odoo/addons/base/models/ir_translation.py

```diff
@@ -500,6 +500,10 @@ def _sync_terms_translations(self, field, records):
                     outdated += translation
                 elif (src, translation.lang) in done:
                     discarded += translation
+                elif any(d.src == src for d in discarded):
+                    # If a previously discarded (because src==value) translation match the close
+                    # term, we should also discard this translation to avoid constraint violation
+                    discarded += translation
```

Comment:
is that correct? don't we drop too many translations? if we make the `unlink` before the `write`, we shouldn't have this issue, right?

Reply:
If the user had a translation for the former with (src != value) it would have been discarded. So a I keep the same behavior